angular-client/src/pages/charging-page/components/pack-voltage/pack-voltage-display/pack-voltage-display.component.ts

```diff
@@ -10,12 +11,27 @@ import { floatPipe } from 'src/utils/pipes.utils';
 })
 export default class PackVoltageDisplay {
   voltage: number = 0;
+  packVoltData: GraphData[] = [];
+  resetGraphButton = {
+    onClick: () => {
+      this.packVoltData = [];
+    },
+    icon: 'restart_alt'
+  };
+  mobileThreshold = 1070;
+  isDesktop = window.innerWidth > this.mobileThreshold;
+
+  @HostListener('window:resize', ['$event'])
+  onResize() {
+    this.isDesktop = window.innerWidth >= this.mobileThreshold;
+  }
 
   constructor(private storage: Storage) {}
 
   ngOnInit() {
     this.storage.get(IdentifierDataType.PACK_VOLTAGE).subscribe((value) => {
       this.voltage = floatPipe(value.values[0]);
+      this.packVoltData.push({ x: decimalPipe(value.time), y: this.voltage });
```

Comment:
don't do decimalPipe, it's just a string that you need to convert to an int, so do this: +value.time (everywhere you're doing this)